Jenkinsfile

```diff
@@ -569,6 +569,46 @@ pipeline {
                  }
              }
         }
+        stage('Self-Driving End-to-End Test') {
+            agent {
+                docker {
+                    image 'noisepage:focal'
+                    args '--cap-add sys_ptrace -v /jenkins/ccache:/home/jenkins/.ccache'
+                }
+            }
+            steps {
+                sh 'echo $NODE_NAME'
+                sh script: 'echo y | sudo ./script/installation/packages.sh all', label: 'Installing packages'
+
+                sh script: '''
+                mkdir build
+                cd build
+                cmake -GNinja -DNOISEPAGE_UNITY_BUILD=ON -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_BUILD_TYPE=Release -DNOISEPAGE_USE_ASAN=OFF -DNOISEPAGE_USE_JEMALLOC=ON -DNOISEPAGE_BUILD_TESTS=OFF  -DNOISEPAGE_BUILD_SELF_DRIVING_TESTS=ON ..
+                ninja mini_runners''', label: 'Self-driving tests (Compile mini_runners)'
+
+                // The parameters to the mini_runners target are (arbitrarily picked to complete tests within a reasonable time / picked to exercise all OUs).
+                // Specifically, the parameters chosen are:
+                // - mini_runner_rows_limit=100, which sets the maximal number of rows/tuples processed to be 100 (small table)
+                // - rerun=0, which skips rerun since we are not testing benchmark performance here
+                // - warm_num=0, which skips the warmup iterations.
```

Comment:
How about setting `warm_num` to 1? This way we can also test the warmup part of the code. I've tried that increasting the rerun iterations would multiply the runtime. But having a warmup iteration does not seem too bad.

Reply:
Oh sure, thought it is just data generation for running the `mini_runners`.